src/test/java/org/pbcsp/SudoRulesTest.java

```diff
@@ -0,0 +1,43 @@
+package org.pbcsp;
+
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+public class SudoRulesTest
+{
+    private CLIPSRunner clips;
+
+    @BeforeClass
+    public void setUp() throws Exception
+    {
+        clips = new CLIPSRunner(Assert::assertEquals);
+    }
+
+    @AfterClass
+    public void tearDown() throws Exception
+    {
+        clips.close();
+    }
+
+    @DataProvider
+    public Object[][] enumerateExamples() throws Exception
+    {
+        return Files.walk(Path.of("Examples/Sudoku-examples/CLIPS-puzzles"))
```

Comment:
This magically discovers the puzzles in the listed directory and turns them into test cases.